src/coreclr/nativeaot/Runtime/wasm/PalRedhawkWasm.cpp

```diff
@@ -191,6 +191,186 @@ extern "C" int __cxa_thread_atexit(Dtor dtor, void* obj, void*)
 #endif // TARGET_WASI
 #endif // !FEATURE_WASM_MANAGED_THREADS
 
+// TODO-LLVM: For now, a copy of the single threaded implementation.
+#ifdef FEATURE_WASM_MANAGED_THREADS
+int __cxa_thread_atexit(void (*func)(), void *obj, void *dso_symbol)
+{
+    return 0;
+}
+
+//
+// Note that we return the native stack bounds here, not shadow stack ones. Currently this functionality is mainly
+// used for RuntimeHelpers.TryEnsureSufficientExecutionStack, and we do use the native stack in codegen, so this
+// is an acceptable approximation.
+//
+extern "C" unsigned char __stack_low;
+extern "C" unsigned char __stack_high;
+
```

Comment:
There is a lot of code copied here and it's not clear what part of it is needed and what part is there because WASI doesn't actually `-pthread` support yet.

Reply:
removed